Lib/typing.py

```diff
@@ -823,6 +874,96 @@ def __init__(self, name, *constraints, bound=None,
             self.__module__ = def_mod
 
 
+class TypeVarTuple(_Final, _Immutable, _root=True):
+
+    def __init__(self, name):
+        self._name = name
+        self._unpacked = UnpackedTypeVarTuple(name, self)
+
+    def __iter__(self):
+        yield self._unpacked
+
+    def __repr__(self):
+        return self._name
+
+
+class UnpackedTypeVarTuple(_Final, _Immutable, _root=True):
+
+    def __init__(self, name, packed):
+        self._name = name
+        self._packed = packed
+        # __parameters__ will be use to figure out what the __parameters__
+        # of e.g. `tuple[*Ts]` should be. In the case of an unpacked
+        # TypeVarTuple, it's unclear whether this should refer to the
+        # UnpackedTypeVarTuple or the TypeVarTuple itself. We somewhat
+        # arbitrarily decide to go for the latter.
+        self.__parameters__ = (packed,)
+
+    def __repr__(self):
+        return '*' + self._name
+
+
+class StarredTuple(_Final, _Immutable, _root=True):
+    """Implementation of starred tuple for older versions of Python.
+
+    A starred tuple is e.g. tuple[*tuple[int, str]]. In newer versions of
```

Comment:
But this implementation is only used in new versions of Python, so why do we need this?